dev/WindowsAppRuntime_MSIXInstallFromPath/main.cpp

```diff
@@ -0,0 +1,303 @@
+
+        other.m_packageId = nullptr;
+        memset(other.m_buffer, 0, sizeof(other.m_buffer));
+    }
+
+    ~PackageId() = default;
+
+    PackageId& operator=(PackageId&& other)
+    {
+        if (this != &other)
+        {
+            m_packageFullName = std::move(other.m_packageFullName);
+            memcpy(m_buffer, other.m_buffer, sizeof(m_buffer));
+            m_packageId = reinterpret_cast<PACKAGE_ID*>(m_buffer);
+
+            other.m_packageId = nullptr;
+            memset(other.m_buffer, 0, sizeof(other.m_buffer));
+        }
+        return *this;
+    }
+
+    const std::wstring& PackageFullName() const
+    {
+        return m_packageFullName;
+    }
+
+    PCWSTR Name() const
+    {
+        return m_packageId->name;
+    }
+
+    PACKAGE_VERSION Version() const
+    {
+        return m_packageId->version;
+    }
+
+    winrt::Windows::System::ProcessorArchitecture Architecture() const
+    {
+        return static_cast<winrt::Windows::System::ProcessorArchitecture>(m_packageId->processorArchitecture);
+    }
+
+    PCWSTR ResourceId() const
+    {
+        return m_packageId->resourceId;
+    }
+
+    PCWSTR PublisherId() const
+    {
+        return m_packageId->publisherId;
+    }
+
+    PCWSTR PackageFamilyName() const
+    {
+        if (m_packageFamilyName.empty())
+        {
+            wchar_t packageFamilyName[PACKAGE_FAMILY_NAME_MAX_LENGTH + 1]{};
+            uint32_t packageFamilyNameLength{ static_cast<uint32_t>(ARRAYSIZE(packageFamilyName)) };
+            THROW_IF_WIN32_ERROR_MSG(::PackageFamilyNameFromFullName(m_packageFullName.c_str(), &packageFamilyNameLength, packageFamilyName), "PackageFullName:%ls", m_packageFullName.c_str());
+            m_packageFamilyName = packageFamilyName;
+        }
+        return m_packageFamilyName.c_str();
+    }
+
+    explicit operator bool() const
+    {
+        return m_packageId != nullptr;
+    }
+
+private:
+    std::wstring m_packageFullName;
+    mutable std::wstring m_packageFamilyName;
+    const PACKAGE_ID * m_packageId{};
+    BYTE m_buffer[sizeof(PACKAGE_ID) + (PACKAGE_NAME_MAX_LENGTH + 1 +
+                                        PACKAGE_RESOURCEID_MAX_LENGTH + 1 +
+                                        PACKAGE_PUBLISHERID_MAX_LENGTH + 1) * sizeof(WCHAR)]{};
+};
+
+void Help()
+{
+    wprintf(L"WindowsAppRuntime_MSIXInstallFromPath [options] <path>\n"
+            L"options:\n"
+            L"  -?, --help = Display help\n"
+            L"          -- = End of options\n"
+            L"where:\n"
+            L"  path = Path where Windows App SDK's MSIX packages can be found\n"
+            L"           => Microsoft.WindowsAppRuntime.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.DDLM.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Main.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Singleton.*.msix\n");
+}
+
+HRESULT JustDoIt(PCWSTR path) noexcept try
+{
+    wprintf(L"path: %s\n", path);
```

Comment:
Debugging printfer will go away in the follow up PR after VS invoking this is tested